core/src/main/kotlin/Util.kt

```diff
@@ -317,48 +331,105 @@ public fun Intents.IntentsBuilder.enableEvent(event: KClass<out Event>): Unit =
     CategoryCreateEvent::class,
     DMChannelCreateEvent::class,
     NewsChannelCreateEvent::class,
+    StageChannelCreateEvent::class,
     StoreChannelCreateEvent::class,
     TextChannelCreateEvent::class,
+    UnknownChannelCreateEvent::class,
     VoiceChannelCreateEvent::class,
 
     ChannelUpdateEvent::class,
     CategoryUpdateEvent::class,
     DMChannelUpdateEvent::class,
     NewsChannelUpdateEvent::class,
+    StageChannelUpdateEvent::class,
     StoreChannelUpdateEvent::class,
     TextChannelUpdateEvent::class,
+    UnknownChannelUpdateEvent::class,
     VoiceChannelUpdateEvent::class,
 
     ChannelDeleteEvent::class,
     CategoryDeleteEvent::class,
     DMChannelDeleteEvent::class,
     NewsChannelDeleteEvent::class,
+    StageChannelDeleteEvent::class,
     StoreChannelDeleteEvent::class,
     TextChannelDeleteEvent::class,
+    UnknownChannelDeleteEvent::class,
     VoiceChannelDeleteEvent::class,
 
-    ChannelPinsUpdateEvent::class,
-    -> {
-        +Guilds
-        +DirectMessages
-    }
+    ThreadChannelCreateEvent::class,
+    NewsChannelThreadCreateEvent::class,
+    TextChannelThreadCreateEvent::class,
+    UnknownChannelThreadCreateEvent::class,
+
+    ThreadUpdateEvent::class,
+    NewsChannelThreadUpdateEvent::class,
+    TextChannelThreadUpdateEvent::class,
+    UnknownChannelThreadUpdateEvent::class,
+
+    ThreadChannelDeleteEvent::class,
```

Comment:
ThreadChannelDeleteEvent is enough here, the other two are sub types

Reply:
This does not work, `when` uses `equals`, from `KClass.equals`:
> Returns true if this KClass instance represents the same Kotlin class as the class represented by other. On JVM this means that all of the following conditions are satisfied:
> 1. other has the same (fully qualified) Kotlin class name as this instance.
> 2. other's backing Class object is loaded with the same class loader as the Class object of this instance.
>
> [...]

Also see this Kotlin Playground: https://pl.kotl.in/H1e13i-bw